src/ngraph/frontend/onnx_import/op/softmax.cpp

```diff
@@ -30,20 +30,20 @@ namespace ngraph
             {
                 NodeVector softmax(const Node& node)
                 {
-                    NodeVector inputs{node.get_ng_inputs()};
-                    auto data = inputs.at(0);
-                    auto data_shape = data->get_shape();
+                    const auto data = node.get_ng_inputs().at(0);
+                    const auto data_rank = data->get_output_partial_shape(0).rank();
 
-                    int axis = node.get_attribute_value<int64_t>("axis", 1);
-                    const auto normalized_axis =
-                        ngraph::normalize_axis(node.get_description(), axis, data_shape.size());
+                    NGRAPH_CHECK(
```

Comment:
Is this only needed for `normalize_axis`? If so, this check shoud not be needed after @mbencer's changes.

Reply:
Yes, this is required for axis normalization which can throw without this check.